src/LockTable.h

```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) 2014-2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef RAMCLOUD_LOCKTABLE_H
+#define RAMCLOUD_LOCKTABLE_H
+
+#include "Common.h"
+
+#include "Atomic.h"
+#include "Fence.h"
+
+namespace RAMCloud {
+
+/**
+ * The LockTable represents a collection a locks with unique IDs.  The locks can
+ * be acquired and released like any typical lock based our their lockId. The
+ * only restriction on the 64-bit lockId is that the value 0 can not be used.
```

Comment:
This last sentence doesn't belong here; It's only needed in the documentation for arguments that must have this property.